kubernetes/deployments_kubeclient_blackbox_test.go

```diff
@@ -0,0 +1,452 @@
+	mock.getter = getter
+	getter.result = mock
+	return mock, nil
+}
+
+type testMetricsGetter struct {
+	config *kubernetes.MetricsClientConfig
+}
+
+type testMetrics struct{}
+
+func (getter *testMetricsGetter) GetMetrics(config *kubernetes.MetricsClientConfig) (kubernetes.MetricsInterface, error) {
+	getter.config = config
+	return testMetrics{}, nil
+}
+
+func (testMetrics) GetCPUMetrics(pods []*v1.Pod, namespace string, startTime time.Time) (*app.TimedNumberTuple, error) {
+	return nil, nil // TODO
+}
+
+func (testMetrics) GetCPUMetricsRange(pods []*v1.Pod, namespace string, startTime time.Time, endTime time.Time,
+	limit int) ([]*app.TimedNumberTuple, error) {
+	return nil, nil // TODO
+}
+
+func (testMetrics) GetMemoryMetrics(pods []*v1.Pod, namespace string, startTime time.Time) (*app.TimedNumberTuple, error) {
+	return nil, nil // TODO
+}
+
+func (testMetrics) GetMemoryMetricsRange(pods []*v1.Pod, namespace string, startTime time.Time, endTime time.Time,
+	limit int) ([]*app.TimedNumberTuple, error) {
+	return nil, nil // TODO
+}
+
+func (testMetrics) GetNetworkSentMetrics(pods []*v1.Pod, namespace string, startTime time.Time) (*app.TimedNumberTuple, error) {
+	return nil, nil // TODO add fake impl when tests exercise this code
+}
+
+func (testMetrics) GetNetworkSentMetricsRange(pods []*v1.Pod, namespace string, startTime time.Time, endTime time.Time,
+	limit int) ([]*app.TimedNumberTuple, error) {
+	return nil, nil // TODO add fake impl when tests exercise this code
+}
+
+func (testMetrics) GetNetworkRecvMetrics(pods []*v1.Pod, namespace string, startTime time.Time) (*app.TimedNumberTuple, error) {
+	return nil, nil // TODO add fake impl when tests exercise this code
+}
+
+func (testMetrics) GetNetworkRecvMetricsRange(pods []*v1.Pod, namespace string, startTime time.Time, endTime time.Time,
+	limit int) ([]*app.TimedNumberTuple, error) {
+	return nil, nil // TODO add fake impl when tests exercise this code
+}
+
+func TestGetMetrics(t *testing.T) {
+	kubeGetter := &testKubeGetter{
+		cmInput: defaultConfigMapInput,
+	}
+	metricsGetter := &testMetricsGetter{}
+
+	token := "myToken"
+	testCases := []struct {
+		clusterURL    string
+		expectedURL   string
+		shouldSucceed bool
+	}{
+		{"https://api.myCluster.url:443/cluster", "https://metrics.myCluster.url", true},
+		{"https://myCluster.url:443/cluster", "", false},
+	}
+
+	for _, testCase := range testCases {
+		config := &kubernetes.KubeClientConfig{
+			ClusterURL:        testCase.clusterURL,
+			BearerToken:       token,
+			UserNamespace:     "myNamespace",
+			KubeRESTAPIGetter: kubeGetter,
+			MetricsGetter:     metricsGetter,
+		}
+		_, err := kubernetes.NewKubeClient(config)
+		if testCase.shouldSucceed {
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			assert.Equal(t, testCase.expectedURL, metricsGetter.config.MetricsURL, "Incorrect Metrics URL")
+			assert.Equal(t, token, metricsGetter.config.BearerToken, "Incorrect bearer token")
+		} else {
+			if err == nil {
```

Comment:
you could call `require.Error(t, err)` instead

Reply:
changed.

Reply:
ok